process_ensembles.py

```diff
@@ -0,0 +1,149 @@
+'''
+MIT License
+
+Copyright (c) 2023 Somayeh Hussaini, Michael Milford and Tobias Fischer
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+'''
+
+import argparse
+import os
+import os.path
+import sys
+from pathlib import Path
+import numpy as np
+
+from tools.logger import Logger
+from non_modular_snn.snn_model_evaluation import compute_recall, invert_dMat 
+
+
+
+
+def main(args):
+    
+    
+    data_path = os.path.join(args.mainfolder_path, f"merged_results_ens_SNN_S{len(args.seeds)}/")
+    Path(data_path).mkdir(parents=True, exist_ok=True)
+    
+    sys.stdout = Logger(data_path, logfile_name=f"logfile_ensembling")
+    
+    summed_rates_all = []
+    n_values = [1, 5, 10, 15, 20, 25]
+    
+    for i in range(len(args.seeds)):
+        
+        print("Seed: ", args.seeds[i])
+        module_path = args.SNN_data_path.format(args.seeds[i], args.epochs[i], args.thresholds[i])
+
+        summed_rates_i_datapath = os.path.join(module_path, "summed_rates.npy")
+        summed_rates_i = np.load(summed_rates_i_datapath)
+        shuffled_indices_path = os.path.join(args.mainfolder_path, f"shuffled_indices_L{args.num_query_imgs}_S{args.seeds[i]}.npy")
+        shuffled_indices = np.load(shuffled_indices_path)
+        summed_rates_i = get_unshuffled_results(summed_rates_i, shuffled_indices, args.num_cal_labels)
+        
+        summed_rates_all.append(summed_rates_i)
+        
+        gt_labels = np.arange(summed_rates_i.shape[0])
+        rates_matrix = invert_dMat(summed_rates_i)
+        inv_test_results_idx = get_pred_results(rates_matrix)
+        numQ = rates_matrix.shape[1]
+        compute_recall(gt_labels, inv_test_results_idx, numQ, n_values, data_path, name=f"recallAtN_SNN{args.seeds[i]}_SL1")
+        
+    print("Ensembling...")
+    summed_rates_all = np.array(summed_rates_all)
```

Comment:
Probably would be good to briefly add comments what those variables are, e.g. explaining what the dimensions are.

Reply:
Added descriptions